src/components/tabs/index.d.ts

```diff
@@ -0,0 +1,39 @@
+/// <reference path="../common.d.ts" />
+
+declare module '@elastic/eui' {
+  import { MouseEventHandler, ReactNode, SFC } from 'react';
+
+  type TAB_SIZES = 's' | 'm';
+
+  type EuiTabProps = CommonProps & {
+    onClick: MouseEventHandler<HTMLButtonElement>;
+    isSelected?: boolean;
+    children?: ReactNode[];
+    disabled?: boolean;
+  };
+
+  type EuiTabsProps = CommonProps & {
+    children?: ReactNode[];
+    size?: TAB_SIZES;
+    expand?: boolean;
+  };
+
+  export interface EuiTabbedContentTab {
+    id: string;
+    name: string;
+    content: ReactNode;
+  }
+
+  type EuiTabbedContentProps = CommonProps & {
+    tabs: EuiTabbedContentTab[];
+    onTabClick?: (tab: EuiTabbedContentTab) => void;
+    selectedTab?: EuiTabbedContentTab;
+    initialSelectedTab?: EuiTabbedContentTab;
+    size?: TAB_SIZES;
+    expand?: boolean;
+  }
+
+  export const EuiTab: SFC<EuiTabProps>;
```

Comment:
Because EUI forwards extra all props to an underlying element, we annotate this further with `HTMLAttributes<HTMLDivElement>`, and solely by convention we usually include `CommonProps` on this line instead of intersecting them together above.

```suggestion
  export const EuiTab: SFC<EuiTabProps & CommonProps & HTMLAttributes<HTMLDivElement>>;
```

which comes with another bonus: `div` elements can optionally have `children` making the explicit `children?` in `EuiTabProps` & `EuiTabsProps` redundant.